sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -1079,6 +1079,13 @@ def construct_bundle_application(self,
                                   ):
     # type: (...) -> beam_fn_api_pb2.BundleApplication
     transform_id, main_input_tag, main_input_coder, outputs = op.input_info
+    # The main_input_coder should be the main_input_coder of
+    # SdfTruncateSizedRestrictions if SdfProcessSizedElements is following
+    # SdfTruncateSizedRestrictions.
+    if (isinstance(op, operations.SdfProcessSizedElements) and
```

Comment:
I don't think this is necessary.

The bundle application should have splits/residuals for the SdfProcessSizedElements and should use its input coder and not the input coder of the sdf_truncate_op. The runner can choose to feed the split back through the truncate operation or it can choose to feed the split directly to the SdfProcessSizedElements based upon how it chooses to construct the graph.

Reply:
The problem here is the restriction coder. At least for `FnApiRunner`, when a `PCollection` crosses the wire, it seems like any non-standard coder will be length-prefixed. Here, the input coder of `SdfTruncateSizedRestrictions ` is length-prefixed, but `SdfProcessSizedElements ` is not. On the `FnApiRunner` side, it decodes the element-restriction pair with the length-prefixed coder.